lib/ansible/plugins/action/reboot.py

```diff
@@ -103,9 +103,14 @@ def get_system_boot_time(self):
         if '1970-01-01 00:00' in command_result['stdout']:
             command_result = self._low_level_execute_command('uptime -s', sudoable=self.DEFAULT_SUDOABLE)
 
+        # This is a last resort for bare Linux systems (e.g. OpenELEC) where 'who -b' or 'uptime -s' are not supported.
+        # Other options like parsing /proc/uptime or default uptime output are less reliable than this
+        if command_result['rc'] != 0:
+            command_result = self._low_level_execute_command('cat /proc/sys/kernel/random/boot_id', sudoable=self.DEFAULT_SUDOABLE)
```

Comment:
wouldn't we need the previous boot id to compare to?

Reply:
@bcoca The comparing is done in `run()`. This method just gets the boot time and returns it.

Reply:
actually .. uptime is a bad check then, since you get a race condition in the case of 'chained reboots'

not something that will happen often, but you can get to the point in which the previous uptime < current uptime even though you rebooted. You should check the uptime vs the time since you started the reboot.

Reply:
The `uptime -s` (and `who -b`) command return a string that is the time the last system was booted, different than what `uptime` is normally used for, which is reporting _total_ system up time.

The plugin doesn't do anything more than look for the output of this string to change. If it has not changed, it assumes the system has not been rebooted.

Since it is not doing `timedelta` comparison on the uptime output, I do not believe we will hit the scenario you describe.

Reply:
i sit corrected then, i should have looked closer at the switches